### PR TITLE
Change addition parameter in InternalVolume from a Real to a Function

### LINKED ISSUE #10626: Change the addition parameter in InternalVolume from a Real to a Function
## Rationale

It would be nice to have the capability to specify an addition to the InternalVolume calculation that varies with time. 

## Description

Change _addition (An additional volume to be included in the internal volume calculation) to accept a function rather than a real.

## Impact

Current tests should still pass as a Real value in the input file would be treated as a constant function.  Increased flexibility.

### PR DESCRIPTION
**Pull request by gambka**
Changes the addition parameter in InternalVolume from a Real to a Function to allow the addition to vary in time.  Existing test cases are not affected.

Closes #10626.

### CODE REVIEW

**Comment by gambka** [comment]
The failure in compiling moose_test-dbg appears to be unrelated to these changes.

File: modules/misc/src/postprocessors/InternalVolume.C [hunk 1]
```diff
@@ -22,7 +23,7 @@ validParams<InternalVolume>()
       "component", 0, "component<3", "The component to use in the integration");
   params.addParam<Real>(
       "scale_factor", 1, "A scale factor to be applied to the internal volume calculation");
-  params.addParam<Real>(
+  params.addParam<FunctionName>(
```
Comment (andrsd) [inline modules/misc/src/postprocessors/InternalVolume.C]: Modify the doco so that it says time-dependent function is expected.

**Review by andrsd — CHANGES_REQUESTED** [review]
Either:

1. update the test so that time-dependent function is used and regold.
2. add new test with a function.

My preference would be 1, but 2 will do as well.

File: modules/misc/src/postprocessors/InternalVolume.C (hunk 1 above)
Comment (gambka) [reply modules/misc/src/postprocessors/InternalVolume.C]: I want the capability to supply a time-dependent function but want to also retain the capability of people supplying just a constant value that is constant in time as is in some of the existing tests.  

I wouldn't say that time-dependent function is expected but more of an option. I can add a second sentence that states "Can vary in time." or something along those lines.

**Comment by gambka** [comment]
As for the test, the existing one is still using the capability however it is supplying a constant value which is treated as a ConstantFunction when read in. In order to test both functionalities if another test is required I think a slightly modified one that varies the addition in time would be better than option 1 because I would like to retain the test of an existing functionality of a constant addition.

File: modules/misc/src/postprocessors/InternalVolume.C (hunk 1 above)
Comment (andrsd) [reply modules/misc/src/postprocessors/InternalVolume.C]: Yes. But, in general functions are time and space dependent. You are expecting just time-dependency - not spacial dependency. We cannot check if the supplied function is time-dependent, so when people supply space-dependent function, they will get constant behavior for point (0, 0, 0). So we need to capture that in the doco.

**Comment by andrsd** [comment]
> constant value which is treated as a ConstantFunction when read in.

Yes, but we want to test that _user supplied_ __time-dependent__ function will produce the right answer. Which constant function does not do.

**Comment by gambka** [comment]
There are currently two tests, one for TM and one for SM that point to the same gold file. I'll update both and regold.

**Review by andrsd — DISMISSED** [review]

**Comment by gambka** [comment]
It appears that I had to rerun the clang format a second time. It appears to have dismissed your review .

**Review by andrsd — APPROVED** [review]

**Pull request merged**